addOns/ascanrules/src/test/java/org/zaproxy/zap/extension/ascanrules/httputils/HtmlContextAnalyserUnitTest.java

```diff
@@ -164,4 +164,14 @@ void shouldNotBreakParsingTagsInInvalidHtml() {
         tagMap = HtmlContextAnalyser.parseTag(tag);
         assertThat(tagMap.size(), is(equalTo(3)));
     }
+
+    @Test
+    void shouldNotRaiseStringIndexOutOfBoundsException() {
+        String tag = "<html> <body> <span>></span> </body> </html>";
```

Comment:
```suggestion
        String tag = "<html> <body> <span>></span> <a></a> </body> </html>";
```
To verify the parsing does not end earlier than needed, browsers keep parsing.